test_app/Gemfile.lock

```diff
@@ -0,0 +1,223 @@
+GEM
```

Comment:
What is this file about?

Reply:
This is the file where bundler will look at when installing the dependencies. The Bundler records the exact versions of dependecies that were installed. So when the same project is loaded on another machine, running bundle will install the exact same versions of dependencies that specified in this file, rather than just using the Gemfile and installing the most recent versions.

Reply:
But, I can set the exact version to be used in Gemfile. Why need to use this?

Reply:
If you set an exact version in the Gemfile then it will stop you from doing `bundle update gem`_ to_update the gem.  The pessimistic versions (~>) or no version specified in gemfile allows you to run` bundle update gem`_to_update and get the latest (minor) version

Reply:
If I use specific version, I don't need to run bundle update. I can just manually update the version in Gemfile

Reply:
Ya but then you would have to keep track of the gem version and manually update every time. When we use ~> and no exact version, it provides the automation for us so we save time.